pkg/reconciler/knativeserving/knativeserving_controller.go

```diff
@@ -164,6 +176,33 @@ func (r *Reconciler) transform(instance *servingv1alpha1.KnativeServing) (mf.Man
 	return r.config.Transform(transforms...)
 }
 
+// Filter the resources
+func (r *Reconciler) filter(manifest *mf.Manifest, instance *servingv1alpha1.KnativeServing) (mf.Manifest, error) {
+	r.Logger.Debug("Filtering manifest")
+	labels := map[string]string{}
+	if !isIngressIstio(manifest, instance) {
+		r.Logger.Debug("Removing istio extention resources")
+		for k, v := range istioLabels {
+			labels[k] = v
+		}
+	}
+	if !isAutoTLSEnabled(manifest, instance) {
+		r.Logger.Debug("Removing autoTLS extention resources")
+		for k, v := range autoTLSLabels {
+			labels[k] = v
+		}
+	}
+	if len(labels) == 0 {
+		return *manifest, nil
+	}
+	// TODO: manifestival's Delete() prints error logs but never returns error actually, so this IsNoMatchError does not make sense.
+	// But leave this code here to show we cannot control "no matches for kind" erorr atm.
+	if err := manifest.Filter(byLabels(labels)).Filter(mf.NotCRDs).Delete(); err != nil && !meta.IsNoMatchError(err) {
+		return *manifest, err
+	}
+	return manifest.Filter(byNoLabels(labels)), nil
```

Comment:
FYI: `byNoLabels(labels)` is equivalent to `mf.Complement(byLabels(labels))`

In v0.3.0, I renamed `mf.Complement` to `mf.None`

There is already a `mf.ByLabel` filter, too, but it only takes a single key/value. It'd be nice to have a `mf.ByLabels` alternative that takes the map.